src/test/java/seedu/duke/command/PinCommandTest.java

```diff
@@ -1,10 +1,78 @@
 package seedu.duke.command;
 
+import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import seedu.duke.data.notebook.Note;
+import seedu.duke.data.notebook.Notebook;
 
 class PinCommandTest {
+    private Note notePinned;
+    private Note noteNotPinned;
+
+    private Notebook notebook;
+
+    private static final String NOTE_CONTENT = "This is a test note.";
+    private static final String NOTE1_TITLE = "TestNote1";
+    private static final String NOTE2_TITLE = "TestNote2";
+
+    @BeforeEach
+    void setUp() {
+        notePinned = new Note(NOTE1_TITLE, NOTE_CONTENT, true);
+        noteNotPinned = new Note(NOTE2_TITLE, NOTE_CONTENT, false);
+
+        notebook = new Notebook();
+
+        notebook.addNote(notePinned);
+        notebook.addNote(noteNotPinned);
+    }
 
     @Test
     void execute() {
     }
+
+    @Test
+    void execute_inputIndex_NoteExists_PinsAndUnpinsNotes() {
+        assertEquals(getExecutionStringInputIndex(notebook, 0),NOTE1_TITLE + " pinned: " + 'N');
+        assertEquals(getExecutionStringInputIndex(notebook, 1),NOTE2_TITLE + " pinned: " + 'Y');
```

Comment:
You should swap the two. assertEquals method arguments are expected and actual. For the sake of debugging in the future it would be better if we caught it now
```suggestion
        assertEquals(NOTE1_TITLE + " pinned: " + 'N', getExecutionStringInputIndex(notebook, 0));
        assertEquals(NOTE2_TITLE + " pinned: " + 'Y', getExecutionStringInputIndex(notebook, 1));
```